### PR TITLE
Add send via paypal button

### PR DESCRIPTION
**Pull request by alphashuro**
#1079

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -92,6 +94,7 @@ group :development, :test do
   gem 'slack-notifier'
   gem 'git'
   gem 'octokit'
+  gem 'dotenv-rails'
```
Comment (alphashuro) [inline Gemfile]: just moved this to `group :development, :test`

**Comment by alphashuro** [comment]
is anyone looking at this?

File: app/controllers/api/v0/cash_outs_controller.rb [hunk 2]
```diff
@@ -3,6 +3,14 @@ class Api::V0::CashOutsController < Api::V0::BaseController
...
 
   def index
+    check_paypal_batch_status()
+
+    getAll()
+
+    render 'api/v2/cash_outs/index'
+  end
+
+  def getAll
```
Comment (wkonkel) [inline app/controllers/api/v0/cash_outs_controller.rb]: Ruby convention is `snake_case` not `camelCase` (i.e. `def get_all`)

File: app/helpers/api/v2/cash_outs_helper.rb [hunk 3]
```diff
@@ -36,4 +39,99 @@ def order!(collection)
...
+
+    threads = []
+
+    # # Paypal only allows making 500 payouts per request
+    # # so we need to iterate through the collection 500 at a time
+    collection.find_in_batches(batch_size: 500) do |batch|
+      threads << Thread.new { send_paypal_batch(batch) }
+    end
+
+    threads.map(&:join)
```
Comment (wkonkel) [inline app/helpers/api/v2/cash_outs_helper.rb]: Let's keep this simple and not use threads (we rarely have 500+ payments pending):
```
collection.find_in_batches(batch_size: 500) do |batch|
  send_paypal_batch(batch)
end
```

**Comment by alphashuro** [comment]
@wkonkel I've added tests and addressed the feedback, please take a look and let me know!

**Comment by alphashuro** [comment]
@wkonkel please let me know when you're ready to review again

**Comment by alphashuro** [comment]
@wkonkel please cast your gaze upon my humble contribution... Have you forsaken me?

**Comment by alphashuro** [comment]
@rappo please can you give some feedback on this

**Comment by albfan** [comment]
@rappo told on slack to provide feedback on this.

@wkonkel  any advice on adding test to this?

**Comment by alphashuro** [comment]
...

**Comment by barkovv** [comment]
@rappo @wkonkel 
Since tests are implemented is any other reason for not to merging this PR? The main problem of Bountysource for workers is huge cashout delays. This PR helps to resolve the problem. I really doesn't understand why this PR isn't merged for such a long time.

**Comment by alphashuro** [comment]
is there even a remote chance that this will ever be looked at again? 

its been in PR status for so long that it even has conflicts now. considering that there is a bounty on it I would have thought that there was going to be more action on this

**Comment by NoNonsense126** [comment]
@alphashuro Sorry about the delay. Canya.io has just acquired bountysource.

I'm part of the new team who will be working with the codebase going forward. We're exploring and getting to know the codebase while working on the roadmap going forward. I understand that it's been quite awhile, but please give us a little more time and we'll try to get to this as soon as we can. Thank you for your patience.

**Comment by NoNonsense126** [comment]
Hey @alphashuro, apologies that this has been delayed for so long. Could you please update this so that it no longer conflicts, and i promise you that I'll look into it with the highest priority.

**Comment by yuriy-yarosh** [comment]
Well, CI failed due to different Ruby versions in `.ruby-version` and `Gemfile`.
I'll check this one soon.

**Pull request closed without merging**